bounce/server/resource/__init__.py

```diff
@@ -70,13 +70,51 @@ def validate(request_cls, response_cls):
             body should match
     """
 
+    def set_defaults(schema, info, parent_key=None, parent_schema=None):
+        """Sets the default values for params and bodies"""
+        for key in schema:
+            value = schema[key]
+            # base case used for recursion
+            if key == 'default':
+                # need to make sure the key is in the schema,
+                # else we'd be putting in keys
+                # that are not valid properties
+                if parent_key in parent_schema:
+                    # if the key is not specified or there's
+                    # an empty string value for the key,
+                    # update info with the default value
+                    if parent_key not in info:
+                        info[parent_key] = value
+                    elif not info[parent_key]:
```

Comment:
Minor: I think you can just collapse this `if ... elif` into
```suggestion
if parent_key not in info or not info[parent_key]:
    ...
```

Reply:
done